core/src/main/scala/kafka/coordinator/PidMetadata.scala

```diff
@@ -28,4 +30,6 @@ private[coordinator] class PidMetadata(val pid: Long) {
     case other: PidMetadata => pid == other.pid && epoch == other.epoch
     case _ => false
   }
+
+  override def hashCode(): Int = Objects.hash(epoch: java.lang.Short, pid: java.lang.Long)
```

Comment:
Seems a bit wasteful to allocate a `Short`, `Long` and array to compute the hash code of two primitives.